deepspeed/pt/deepspeed_light.py

```diff
@@ -1029,10 +1023,10 @@ def buffered_allreduce_fallback(self, grads=None, elements_per_buffer=500000000)
                 # rank is reducing the same size. In some cases it may make
                 # sense in the future to support the ability to average not
                 # w.r.t. world size but with a different value.
-                param.grad = torch.zeros(param.size(),
-                                         dtype=param.dtype,
-                                         device=param.device)
-                grads.append(param.grad.data)
+                grads.append(
+                    torch.zeros(param.size(),
+                                dtype=param.dtype,
+                                device=param.device))
```

Comment:
Uh oh, I think this reverts #316. Was this an accident or was it causing an issue?

Reply:
Thanks for catching this. It was accidental, probably due to a bad merge. I will look further.